### DISCUSSION TITLE
About that organization in: Rol-Users, User-Accounts, Radio-stations

### DISCUSSION THREAD

**Discussion by draven3166** [AzuraCast/AzuraCast#7539] [Ideas]
Understand that organization and workflow in azuracast panel administration in management. Please correct if the explanation of the current workflow with roles, user accounts and radio stations is wrong.

My **workflow experience** in azuracast: to manage a radio station at level user distinct of superuser first neded create a rol to relation rol with user and in role can configure relation of radio station and manage access to configure or management in this level.
This workflow neded apply with different radio stations and user accounts but its necessary create different roles-user to manage user and radio stations.
In my developer position i see very difficult this workflow to manage users and radio station with Roles user. I consider share my experience with azuracast community explaining one posible extensible and powerfull workflow about Roles, Users, and radio station, explain a relation in this tree entitys to work more scalable.
**Role-user**
It must be configurable and assign the most important configurations within the administration at the level of statistical reports, information and radio administration to have the option of opening or closing access or not to the type of role that is being created. For example, a Reseller ROL will need global control, but an Owner-radio ROL should limit its access specifically to its radio, the same case for the Radio-manager Role which would be below its accesses from the Owner-Radio Role
**User:**
The name of the entity says it all, it is to be able to access the genuine software to be able to manage it, draw reports, or have higher functions always depending on the assigned role level.
**Radio stations:**
At the time of instantiating them, they have all the detailed information and configuration in the azuracast administrative panel.

**IDEA How to relate a user to a Radio station:**
As we mentioned, user roles would define the level of access to the system in general.
To create a dependency relationship between User and Radio station it should be from there A user should be Related or attached to a radio station.
We assume the role assigned to the user is according to our somewhat generic workflow and not do it as in the current azuracast flow, thereby avoiding creating N-user roles for N-radio stations.
We improve the level of user access control by delimiting their access at the azuracast entity administration level, as well as the access level and radio configuration.
With this we achieve immediacy new radio station created with the necessary configurations well.
New user created, assign a role and automatically inherit the functional depth he or she will have.
The most important thing is to relate this user to the radio station that corresponds to their already inherited access.
We avoid creating so many roles to relate radios-users. 

I await your feed to be able to exchange ideas, however the community is for learning in both directions, and open mindsets. 
Greetings, azuracast development team.

**Comment by BusterNeece** [comment]
I don't know that we necessarily want to fully overhaul our roles/actions system, as it's quite powerful and allows power users to do a lot of impressive things with user management...

However, something I've considered is adding the ability to create a "station manager" user simultaneous to the creation of a new station. That is, if you're on the "New Station" modal, you can create a username/password for an account that will automatically be given a new role that gives it full administrative permissions to that one station.

That would greatly simplify the workflow, while still allowing power users to drill into the roles/permissions more specifically.

Thoughts?

**Comment by draven3166** [comment]
Creating user-station simultaneously will increase station control and improve the administrative experience, good feedback, thank you.
**Thought**
Part of being part of a great work team or community is knowing its philosophy, where the thinking is, and the interaction or exchange of ideas, the way of making software enriches the experience as a developer and also as a user. Thank you
